src/App.tsx

```diff
@@ -2,45 +1,87 @@
+//import classNames from 'classnames';
 
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
-import { PostsList } from './components/PostsList';
-import { PostDetails } from './components/PostDetails';
+//import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
+import { useEffect, useState } from 'react';
+import { User } from './types/User';
+import { getUsers } from './api/users';
 import { Loader } from './components/Loader';
+import { PostsList } from './components/PostsList/PostsList';
+import { Post } from './types/Post';
+import { getPosts } from './api/posts';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [notification, setNotification] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  // #region load
+  useEffect(() => {
+    getUsers().then(setUsers);
+  }, []);
 
-              <Loader />
+  useEffect(() => {
+    if (selectedUser) {
+      setIsLoading(true);
+      setPosts([]);
+      setNotification('');
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+      getPosts(selectedUser.id)
+        .then(res => {
+          if (res.length === 0) {
+            setNotification('is-warning');
```

Comment:
The notification state is set to 'is-warning' when there are no posts. Ensure that this aligns with your application's design and user experience goals.